packages/toast/src/use-toast.tsx

```diff
@@ -196,6 +200,24 @@ export function createStandaloneToast({
     })
   }
 
+  toastImpl.promise = (
+    promise: Promise<any>,
+    {
+      success,
+      error,
+      loading,
+    }: {
+      success: UseToastPromiseOption
+      error: UseToastPromiseOption
+      loading: UseToastPromiseOption
+    },
+  ) => {
+    const id = toastImpl({ ...loading, status: "loading", duration: null })
+    promise
+      .then(() => toastImpl.update(id, { ...success, status: "success" }))
+      .catch(() => toastImpl.update(id, { ...error, status: "error" }))
```

Comment:
```suggestion
      .then((data) => toastImpl.update(id, { ...success, status: "success", data }))
      .catch((error) => toastImpl.update(id, { ...error, status: "error", data: error }))
```

I think we need to provide the users will the resolved data or error so they can update the toast's message accordingly.